api/src/main/java/jakarta/mail/Multipart.java

```diff
@@ -266,4 +269,27 @@ public synchronized Part getParent() {
     public synchronized void setParent(Part parent) {
         this.parent = parent;
     }
+
+    protected StreamProvider provider() {
```

Comment:
The writeTo and parse methods both throw MessagingException so I think it would make sense to add throws `MessagingException` here.  We can then wrap `ServiceConfigurationError | RuntimeException` in a MessagingException that the caller is ready to receive.

Reply:
Done